File: samples/snippets/visualbasic/VS_Snippets_Winforms/System.ComponentModel.ITypedList/VB/Customer.vb
' <snippet10>
Imports System.ComponentModel

Public Class Customer
    Implements INotifyPropertyChanged

    Public Sub New()

    End Sub

    Public Sub New(ByVal id As Integer, ByVal name As String, ByVal company As String, ByVal address As String, ByVal city As String, ByVal state As String, ByVal zip As String)
        Me._id = id
        Me._name = name
        Me._company = company
        Me._address = address
        Me._city = city
        Me._state = state
        Me._zip = zip

    End Sub

#Region "Public Properties"

    Private _id As Integer
    Public Property ID() As Integer
        Get
            Return _id
        End Get
        Set(ByVal value As Integer)
            If _id <> value Then
                _id = value
                OnPropertyChanged(New PropertyChangedEventArgs("ID"))
            End If
        End Set
    End Property

    Private _name As String

    Public Property Name() As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            If _name <> value Then
                _name = value
                OnPropertyChanged(New PropertyChangedEventArgs("Name"))
            End If
        End Set
    End Property

    Private _company As String

    Public Property Company() As String
        Get
            Return _company
        End Get
        Set(ByVal value As String)
            If _company <> value Then
                _company = value
                OnPropertyChanged(New PropertyChangedEventArgs("Company"))
            End If
        End Set
    End Property


    Private _address As String

    Public Property Address() As String
        Get
            Return _address
        End Get
        Set(ByVal value As String)
            If _address <> value Then
                _address = value
                OnPropertyChanged(New PropertyChangedEventArgs("Address"))
            End If
        End Set
    End Property


    Private _city As String

    Public Property City() As String
        Get
            Return _city
        End Get
        Set(ByVal value As String)
            If _city <> value Then
                _city = value
                OnPropertyChanged(New PropertyChangedEventArgs("City"))
            End If
        End Set
    End Property


    Private _state As String

    Public Property State() As String
        Get
            Return _state
        End Get
        Set(ByVal value As String)
            If _state <> value Then
                _state = value
                OnPropertyChanged(New PropertyChangedEventArgs("State"))
            End If
        End Set
    End Property

    Private _zip As String

    Public Property ZipCode() As String
        Get
            Return _zip
        End Get
        Set(ByVal value As String)
            If _zip <> value Then
                _zip = value
                OnPropertyChanged(New PropertyChangedEventArgs("ZipCode"))
            End If
        End Set
    End Property

#End Region


#Region "INotifyPropertyChanged Members"

    Public Event PropertyChanged(ByVal sender As Object, ByVal e As System.ComponentModel.PropertyChangedEventArgs) Implements System.ComponentModel.INotifyPropertyChanged.PropertyChanged

    Protected Overridable Sub OnPropertyChanged(ByVal e As PropertyChangedEventArgs)
        RaiseEvent PropertyChanged(Me, e)
    End Sub

#End Region

End Class
' </snippet10>

File: samples/snippets/csharp/VS_Snippets_Winforms/System.ComponentModel.ITypedList/CS/Customer.cs
// <snippet10>
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel;

namespace ITypedListCS
{
	class Customer : INotifyPropertyChanged
    {
        public Customer() {}

        public Customer(int id, string name, string company, string address, string city, string state, string zip)
        {
            this._id = id;
            this._name = name;
            this._company = company;
            this._address = address;
            this._city = city;
            this._state = state;
            this._zip = zip;
        }

        #region Public Properties

        private int _id;

		public int ID
		{
			get { return _id; }
			set
			{
                if (_id != value)
                {
                    _id = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("ID"));
                }
			}
		}

		private string _name;

		public string Name
		{
			get { return _name; }
			set
			{
                if (_name != value)
                {
                    _name = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("Name"));
                }
			}
		}

		private string _company;

		public string Company
		{
			get { return _company; }
			set
			{
                if (_company != value)
                {
                    _company = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("Company"));
                }
			}
		}

		private string _address;

		public string Address
		{
			get { return _address; }
			set
			{
                if (_address != value)
                {
                    _address = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("Address"));
                }
			}
		}

		private string _city;

		public string City
		{
			get { return _city; }
			set
			{
                if (_city != value)
                {
                    _city = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("City"));
                }
			}
		}

		private string _state;

		public string State
		{
			get { return _state; }
			set
			{
                if (_state != value)
                {
                    _state = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("State"));
                }
			}
		}

		private string _zip;

		public string ZipCode
		{
			get { return _zip; }
			set
			{
                if (_zip != value)
                {
                    _zip = value;
                    OnPropertyChanged(new PropertyChangedEventArgs("ZipCode"));
                }
			}
        }

        #endregion

		#region INotifyPropertyChanged Members

		public event PropertyChangedEventHandler PropertyChanged;

		protected virtual void OnPropertyChanged(PropertyChangedEventArgs e)
		{
			if (null != PropertyChanged)
			{
				PropertyChanged(this, e);
			}
		}

		#endregion
	}
}
// </snippet10>